compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_terminator_gen.bal

```diff
@@ -513,7 +524,47 @@ type TerminatorGenerator object {
             if (lhsVar is bir:VariableDcl) {
                 generateVarStore(self.mv, lhsVar, currentPackageName, lhsIndex);
             }
-        }    
+        }
+
+        self.genYieldCheck(fpCall.thenBB, funcName);   
+    }
+
+    function submitToScheduler(bir:VarRef? lhsOp) {
+        bir:BType? futureType = lhsOp.typeValue;
+        boolean isVoid;
+        if (futureType is bir:BFutureType) {
+            isVoid = futureType.returnType is bir:BTypeNil;
+        } else {
+            error err = error( "Invalid return type for async function pointer call" +
```

Comment:
Can there be a case where the execution goes to this else case? Is this not checked at compile time ?

Reply:
It cannot go to else. Did this to avoid not initialized error. 
Will put a default value instead.